core/src/main/scala/org/http4s/ContentCoding.scala

```diff
@@ -18,54 +18,117 @@
-  type Value = ContentCoding
+  override def render(writer: Writer): writer.type =
+    ContentCoding.http4sInstancesForContentCoding.render(writer, this)
+}
 
-  implicit def fromKey(k: CaseInsensitiveString): ContentCoding = ContentCoding(k)
+object ContentCoding {
+  def unsafeFromString(coding: String): ContentCoding =
+    fromString(coding).valueOr(throw _)
 
-  implicit def fromValue(v: ContentCoding): CaseInsensitiveString = v.coding
+  def fromString(coding: String): ParseResult[ContentCoding] =
+    parse(coding)
 
-  val `*` : ContentCoding = registerKey("*".ci)
+  val `*` : ContentCoding = new ContentCoding("*")
 
   // http://www.iana.org/assignments/http-parameters/http-parameters.xml#http-parameters-1
-  val compress = registerKey("compress".ci)
-  val deflate = registerKey("deflate".ci)
-  val exi = registerKey("exi".ci)
-  val gzip = registerKey("gzip".ci)
-  val identity = registerKey("identity".ci)
-  val `pack200-gzip` = registerKey("pack200-gzip".ci)
+  val compress = new ContentCoding("compress")
+  val deflate = new ContentCoding("deflate")
+  val exi = new ContentCoding("exi")
+  val gzip = new ContentCoding("gzip")
+  val identity = new ContentCoding("identity")
+  val `pack200-gzip` = new ContentCoding("pack200-gzip")
 
   // Legacy encodings defined by RFC2616 3.5.
-  val `x-compress` = register("x-compress".ci, compress)
-  val `x-gzip` = register("x-gzip".ci, gzip)
+  val `x-compress` = compress
+  val `x-gzip` = gzip
+
+  def standard: Map[String, ContentCoding] =
+    List(`*`, compress, deflate, exi, gzip, identity, `pack200-gzip`).map(c => c.coding -> c).toMap
+
+  /**
+    * Parse a Transfer Coding
+    */
+  def parse(s: String): ParseResult[ContentCoding] =
+    new Http4sParser[ContentCoding](s, "Invalid Content Coding") with ContentCodingParser {
+      def main = EncodingRangeDecl
+    }.parse
+
+  private[http4s] trait ContentCodingParser extends QValueParser { self: PbParser =>
+
+    def EncodingRangeDecl: Rule1[ContentCoding] = rule {
+      (EncodingRangeDef ~ QualityValue) ~> { (coding: ContentCoding, q: QValue) =>
+        if (q === org.http4s.QValue.One) coding
+        else coding.withQValue(q)
+      }
+    }
+
+    def EncodingRangeDef: Rule1[ContentCoding] = rule {
+      "*" ~ push(ContentCoding.`*`) | Token ~> { s: String =>
+        ContentCoding.standard.getOrElse(s, new ContentCoding(s))
+      }
+    }
+  }
+
+  implicit val http4sInstancesForContentCoding: Show[ContentCoding] with HttpCodec[ContentCoding] =
+    new Show[ContentCoding] with HttpCodec[ContentCoding] {
+      override def show(s: ContentCoding): String = s.toString
+
+      override def parse(s: String): ParseResult[ContentCoding] =
+        ContentCoding.parse(s)
+
+      override def render(writer: Writer, coding: ContentCoding): writer.type =
+        writer << coding.coding.toLowerCase << coding.qValue
+
+    }
+
+  implicit val order: Order[ContentCoding] = Order.by(c => (c.coding.toLowerCase, c.qValue))
```

Comment:
Any reason order is split out instead of part of the instances?

Reply:
I don't know how could I make it part of `http4sInstances` and use `Order.by` at the same time

Reply:
Oh, I see.  I've probably just been defining it myself elsewhere.  If it stands alone, it should follow the naming pattern.  We want to get all our implicits with a unique name to avoid shadowing.

Reply:
What's the naming convention? I'm not sure I've seen it

Reply:
`http4s{nameOfTypeclass}For{nameOfTypeParameter}` is consistent with what Cats does. Or `http4sInstancesFor{nameOfTypeParameter}` for one big compound instance.